### DISCUSSION TITLE
Failed to load dynlib/dll 'libsnap7-21acf2c4.so'

### DISCUSSION THREAD

**Discussion by dengxuankun** [gijzelaerr/python-snap7#412] [Q&A] [answered]
When i ues pyinstaller to pack project,I meet this problem: PyInstallerImportError: Failed to load dynlib/dll 'libsnap7-21acf2c4.so'. Most likely this dynlib/dll was not found when the application was frozen.But I did not find this lib.

**Comment by gijzelaerr** [comment]
why don't you use the binary wheels?

I don't think this is an issue with python-snap7, but a problem with pyinstaller or the way you use it. Therefore i'm moving this to the Q&A section.

**Reply by MAKOMO** [reply]
Same issue here using pip installed binary wheels for aarch64 from  on Raspian OS bookworm (RPi4) under Python 3.11 and latest pyinstaller 6.1. The string '21acf2c4' is found in `libsnap7.so`. 

I get the same error message if I place the libsnap7.so at the top level of my app bundle (now within the subdirectory `_internal` with pyinstaller 6.x). If I rename `libsnap7.so` in the build app to `libsnap7-21acf2c4.so` the app works. Better, if I place the shared lib in `_internal/snap7/lib` it can be loaded without that funny renaming which might not be stable.

I did put the following in my `.spec`file:

```
import os
from PyInstaller.utils.hooks import get_package_paths
BINARIES = [(os.path.join(get_package_paths('snap7')[1], 'lib/libsnap7.so'), 'snap7/lib' )]

a = Analysis(['script.py'],
             binaries=BINARIES,
...
```

**Comment by gijzelaerr** [comment, marked as answer]
Closing as resolved. This was a PyInstaller bundling issue. Python-snap7 3.0 is now a pure Python implementation and no longer requires a native .dll/.so library, making this issue moot.
